leetcode/0003.Longest-Substring-Without-Repeating-Characters/3. Longest Substring Without Repeating Characters.go

```diff
@@ -26,25 +26,35 @@ func lengthOfLongestSubstring(s string) int {
 	return result
 }
 
-// 解法二 滑动窗口
-func lengthOfLongestSubstring_(s string) int {
-	if len(s) == 0 {
-		return 0
+// 解法二 滑动窗口-数组桶
+func lengthOfLongestSubstring1(s string) int {
+	right, left, res := 0, 0, 0
+	var m [256]int
+	for left < len(s) {
+		tmp := m[s[left]-'a']
+		if tmp >= right {
+			right = tmp + 1
+		}
+		m[s[left]-'a'] = left
+		left++
+		res = max(res, left-right)
 	}
-	var freq [256]int
-	result, left, right := 0, 0, -1
+	return res
+}
 
+// 解法二 滑动窗口-哈希桶
+func lengthOfLongestSubstring2(s string) int {
+	right, left, res := 0, 0, 0
+	m := make(map[byte]int, len(s))
```

Comment:
Is it better to name variable m to indexes?

Reply:
done